src/streamlink/plugins/artetv.py

```diff
@@ -45,44 +44,21 @@ class ArteTV(Plugin):
     def can_handle_url(cls, url):
         return _url_re.match(url)
 
-    def _create_stream(self, stream, language):
-        stream_name = "{0}p".format(stream["height"])
-        stream_type = stream["mediaType"]
-        stream_url = stream["url"]
-        stream_language = stream["versionShortLibelle"]
-
-        if language == "de":
-            language = ["DE", "VOST-DE", "VA", "VOA", "Dt. Live", "OV", "OmU"]
-        elif language == "en":
-            language = ["ANG", "VOST-ANG"]
-        elif language == "es":
-            language = ["ESP", "VOST-ESP"]
-        elif language == "fr":
-            language = ["FR", "VOST-FR", "VF", "VOF", "Frz. Live", "VO", "ST mal"]
-        elif language == "pl":
-            language = ["POL", "VOST-POL"]
-
-        if stream_language in language:
-            if stream_type in ("hls", "mp4"):
-                if urlparse(stream_url).path.endswith("m3u8"):
+    def _create_stream(self, streams):
+        variant, variantname = min([(stream["versionProg"], stream["versionLibelle"]) for stream in streams.values()],
+                                   key=itemgetter(0))
+        log.debug("Using the '{0}' stream variant".format(variantname))
```

Comment:
error with py2 because of `Français`

```
$ streamlink https://www.arte.tv/fr/direct/ -l info
src/streamlink_cli/main.py:760: PythonDeprecatedWarning: Python 2.7 ...
[cli][info] Found matching plugin artetv for URL https://www.arte.tv/fr/direct/
error: 'ascii' codec can't encode character u'\xe7' in position 11: ordinal not in range(128)
```

Reply:
Good spot. This should fix it. 
```suggestion
        log.debug(u"Using the '{0}' stream variant".format(variantname))
```

Reply:
Fixed it on the error log too.